### PR TITLE
Cache bypass in dedicated gateway request options

### PR DESCRIPTION
**Pull request by balalakshmin**
# Description

This is to add a new DedicatedGatewayRequestOption "BypassIntegratedCache".
For customers to have more control over integrated cache, we're introducing a new "RequestOption" called "BypassIntegratedCache". This option will allow the customer to decide whether to use integrated cache for each request or not. If this value is set to true, the item/query will be served from backend and won't be cached in Dedicated Gateway.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by sourabh1007 — APPROVED** [review]

**Review by kushagraThapar — CHANGES_REQUESTED** [review]
can be use primitive boolean?
Also, blocking on testing of this PR, once tested, then we can go ahead with the merge.

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/DedicatedGatewayRequestOptions.java [hunk 1]
```diff
@@ -11,6 +11,8 @@ public final class DedicatedGatewayRequestOptions {
 
     private Duration maxIntegratedCacheStaleness;
 
+    private Boolean bypassIntegratedCache;
```
[the remark this answers is not in the document]
Comment (balalakshmin) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/DedicatedGatewayRequestOptions.java:14]: in .Net we are using an optional boolean field.  To follow the same contract I've used a `Boolean` here

[the remark this answers is not in the document]
Comment (balalakshmin) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/DedicatedGatewayRequestOptions.java]: Verified with feature team and changed this to primitive boolean

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/DedicatedGatewayRequestOptions.java [hunk 2]
```diff
@@ -49,4 +51,28 @@ public DedicatedGatewayRequestOptions setMaxIntegratedCacheStaleness(Duration ma
...
+
+    /**
+     * Sets if integrated cache should be enabled or bypassed for the request in Azure CosmosDB service.
+     *
+     * <p>Default value is false</p>
+     *
+     * @param bypassIntegratedCache Boolean value
+     * @return this DedicatedGatewayRequestOptions
+     */
+    public DedicatedGatewayRequestOptions setBypassIntegratedCache(Boolean bypassIntegratedCache) {
```
[the remark this answers is not in the document]
Comment (balalakshmin) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/DedicatedGatewayRequestOptions.java]: Verified with feature team and changed this to primitive boolean

File: sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/implementation/RequestHeadersSpyWireTest.java [hunk 3]
```diff
@@ -216,6 +230,32 @@ public void readItemWithMaxIntegratedCacheStalenessInNegative() {
...
+
+        String documentLink = getDocumentLink();
+
+        client.clearCapturedRequests();
+
+        RequestOptions requestOptions = ModelBridgeInternal.toRequestOptions(cosmosItemRequestOptions);
+        requestOptions.setPartitionKey(new PartitionKey(DOCUMENT_ID));
+        ResourceResponse<Document> response = client.readDocument(documentLink, requestOptions).block();
+        if (cacheBypass) {
+            String responseHeader = response.getResponseHeaders().get("x-ms-cosmos-cache-bypass");
```
Comment (jeet1995) [inline sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/implementation/RequestHeadersSpyWireTest.java:250]: Synced up offline with @balalakshmin. `x-ms-cosmos-cache-bypass` response from the service / gateway denotes whether the request hit the cache or not.

**Review by jeet1995 — APPROVED** [review]
LGTM - thanks for the change.

**Review by kushagraThapar — APPROVED** [review]

**Pull request merged**